### PR TITLE
mv demo nbs into src folder and add fn for cloning to current dir

### PR DESCRIPTION
**Pull request by KPostOffice**
# Issue link
https://issues.redhat.com/browse/RHOAIENG-10266

# What changes have been made
Be not afraid of the many files being changed, they are all just being moved from the root to be inside the sdk src dir

# Verification steps

1. build package via `poetry build`
2. create venv and activate it
3. install built wheel by doing `pip install `<path-to-sdk-dir>/dist/codeflare_sdk-0.0.0.dev0-py3-none-any.whl`
4. activate python shell
5. run `import codeflare_sdk`
6. run `codeflare_sdk.clone_demo_nbs()`
7. verify that `demo-notebooks` dir is created with all demo nbs present

## Checks
- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [x] Manual tests
   - [ ] Testing is not required for this change

### CODE REVIEW

File: src/codeflare_sdk/utils/demos.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import pathlib
+import shutil
+
+package_dir = pathlib.Path(__file__).parent.parent.resolve()
+demo_dir = f"{package_dir}/demo-notebooks"
+
+
+def clone_demo_nbs(dir: str = "./demo-notebooks"):
```
Comment (Ygnas) [inline src/codeflare_sdk/utils/demos.py:8]: Just a thought what if there would be an `overwrite` arg  allowing the directory to be overwritten?

**Review by Ygnas — APPROVED** [review]
Works as expected.

**Review by Bobbins228 — APPROVED** [review]
/approve
Works as expected awesome work on this Kevin!

**Comment by Bobbins228** [comment]
@KPostOffice had to remove approval just had a thought. We should document this in our README

**Review by astefanutti — COMMENTED** [review]
Would a symlink work?

**Comment by KPostOffice** [comment]
> Would a symlink work?

@astefanutti I think copies make more sense that way users always have access to the original demo notebooks in case they adjust it and make some changes that cause it to break

File: src/codeflare_sdk/utils/demos.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import pathlib
+import shutil
+
+package_dir = pathlib.Path(__file__).parent.parent.resolve()
+demo_dir = f"{package_dir}/demo-notebooks"
+
+
+def clone_demo_nbs(dir: str = "./demo-notebooks", overwrite: bool = False):
```
Comment (astefanutti) [inline src/codeflare_sdk/utils/demos.py:8]: What about `copy_demo_notebooks`? Clone gives too much impression it's a Git operation.

File: README.md [hunk 3]
```diff
@@ -7,8 +7,8 @@ An intuitive, easy-to-use python interface for batch resource requesting, access
 
 For guided demos and basics walkthroughs, check out the following links:
 
-- Guided demo notebooks available [here](https://github.com/project-codeflare/codeflare-sdk/tree/main/demo-notebooks/guided-demos), and copies of the notebooks with [expected output](https://github.com/project-codeflare/codeflare-sdk/tree/main/demo-notebooks/guided-demos/notebook-ex-outputs) also available
-- Note that these notebooks will work with the latest `codeflare-sdk` PyPI release. For testing and experimentation with `main` branch, please use the [preview notebooks](https://github.com/project-codeflare/codeflare-sdk/tree/main/demo-notebooks/guided-demos/preview_nbs)
+- Guided demo notebooks available [here](https://github.com/project-codeflare/codeflare-sdk/tree/main/src/demo-notebooks/guided-demos), and copies of the notebooks with [expected output](https://github.com/project-codeflare/codeflare-sdk/tree/main/src/demo-notebooks/guided-demos/notebook-ex-outputs) also available
+- these demos can be copied into your current working directory when using the `codeflare-sdk` by using the `codeflare_sdk.clone_demo_nbs()` function
```
Comment (Bobbins228) [inline README.md:11]:
```suggestion
- these demos can be copied into your current working directory when using the `codeflare-sdk` by using the `codeflare_sdk.copy_demo_nbs()` function
```

Comment (ChughShilpa) [inline README.md:11]: Replace clone with copy

File: src/codeflare_sdk/utils/demos.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+package_dir = pathlib.Path(__file__).parent.parent.resolve()
+demo_dir = f"{package_dir}/demo-notebooks"
+
+
+def copy_demo_nbs(dir: str = "./demo-notebooks", overwrite: bool = False):
+    """
+    Copy the demo notebooks from the package to the current working directory
+
+    overwrite=True will overwrite any files that exactly match files written by clone_demo_nbs in the target directory.
```
Comment (ChughShilpa) [inline src/codeflare_sdk/utils/demos.py:12]: overwrite=True will overwrite any files that exactly match files written by copy_demo_nbs in the target directory.

Comment (ChughShilpa) [inline src/codeflare_sdk/utils/demos.py:12]:
```suggestion
    overwrite=True will overwrite any files that exactly match files written by copy_demo_nbs in the target directory.
```

File: README.md (hunk 3 above)
Comment (ChughShilpa) [reply README.md:11]: @Bobbins228 you won by few seconds ;)

Comment (Bobbins228) [reply README.md]: Fastest gun in the west :)

**Review by Bobbins228 — COMMENTED** [review]
/lgtm

**Pull request merged**